service/models.py

```diff
@@ -18,10 +18,85 @@ def init_db(app):
     """ Initializes the SQLAlchemy app """
     Customer.init_db(app)
 
-
 class DataValidationError(Exception):
     """ Used for an data validation errors when deserializing """
 
+class Address(db.Model):
+    """
+    Class that represents a Address
+    """
+
+    ################
+    # Address Schema
+    ################
+
+    logger = logging.getLogger('flask.app')
+    app = None
+
+    # Table Schema
+    address_id = db.Column(db.Integer, primary_key=True)
+    street = db.Column(db.String(255), nullable=False)
+    city = db.Column(db.String(255), nullable=False)
+    state = db.Column(db.String(255), nullable=False)
+    country = db.Column(db.String(255), nullable=False)
+    pin_code = db.Column(db.String(255), nullable=False)
+    customer_id = db.Column(db.Integer, db.ForeignKey('customer.id', ondelete="CASCADE"), nullable=False)
+
+    def __repr__(self):
+        return f"<Address {self.street} id=[{self.address_id}] customer[{self.id}]>"
+    
+    def __str__(self):
+        return f"{self.street}, {self.city}, {self.state}, {self.country}, {self.pin_code}"
+
+    def serialize(self):
+        """ 
+        Serializes an Address into a dictionary
+        """
+        return {"address_id": self.address_id,
+                "street": self.street,
+                "city": self.city,
+                "state": self.state,
+                "country": self.country,
+                "pin_code": self.pin_code,
+                "customer_id": self.customer_id
+                }
+
+    def deserialize(self, data):
+        """
+        Deserializes an Address from a dictionary
+
+        Args:
+            data (dict): A dictionary containing the Address data
+        """
+        try:
+            self.street = data['street']
+            self.city = data['city']
+            self.state = data['state']
+            self.country = data['country']
+            self.pin_code = data['pin_code']
+            self.customer_id = data['customer_id']
+        except KeyError as error:
+            raise DataValidationError("Invalid Address: missing " + error.args[0]) from error
+        except TypeError as error:
+            raise DataValidationError("Invalid pet: Body of request contained bad or no data " + str(error)) from error
+        return self
+
+    def create(self):
+        """
+        Creates an Address in the database
+        """
+        Address.logger.info('Creating %s', self.street)
+        if not self.address_id:
+            db.session.add(self)
+        db.session.commit()
+        logger.info("Address is saved successfully")
+
+    def delete(self):
+        """ Removes a Address from the database """
+        logger.info("Deleting %s, %s", self.street,self.city)
+        db.session.delete(self)
+        db.session.commit()
+
```

Comment:
Need to add update function for address, 
```
    def update(self):
        """
        Updates an Address to the database
        """
        logger.info("Saving Address ID %s", self.id)
        if not self.id:
            raise DataValidationError("Update called with empty ID field")
        db.session.commit()
```

Reply:
I am having issues with models.py code coverage after adding update function, checking it.

Reply:
Closing as we added the Update function after ensuring relevant code coverage.